flight/targets/brainre1/fw/pios_board.c

```diff
@@ -520,6 +541,94 @@ void PIOS_Board_Init(void) {
 #endif /* PIOS_INCLUDE_BMI160 */
 #endif /* PIOS_INCLUDE_SPI */
 
+	uint8_t hw_i2c;
+	HwBrainRE1I2CPortGet(&hw_i2c);
+
+	if (hw_i2c != HWBRAINRE1_I2CPORT_DISABLED) {
+#if defined(PIOS_INCLUDE_I2C)
+		if (PIOS_I2C_Init(&pios_i2c_external_id, &pios_i2c_external_cfg)) {
+			PIOS_DEBUG_Assert(0);
+			AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_CRITICAL);
+		}
+		if (PIOS_I2C_CheckClear(pios_i2c_external_id) != 0) {
+			AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_CRITICAL);
+		}
+
+		if (AlarmsGet(SYSTEMALARMS_ALARM_I2C) == SYSTEMALARMS_ALARM_UNINITIALISED)
+				AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_OK);
+#endif // PIOS_INCLUDE_I2C
+	}
+
+	PIOS_WDG_Clear();
+
+	uint8_t hw_ext_baro;
+	HwBrainRE1ExtBaroGet(&hw_ext_baro);
+
+	if (((hw_i2c == HWBRAINRE1_I2CPORT_EXTBARO) || (hw_i2c == HWBRAINRE1_I2CPORT_EXTBAROANDEXTMAG)) &&
+	    (hw_ext_baro == HWBRAINRE1_EXTBARO_BMP280)) {
+#if defined(PIOS_INCLUDE_I2C)
+#if defined(PIOS_INCLUDE_BMP280)
+		if (PIOS_BMP280_Init(&pios_bmp280_cfg, pios_i2c_external_id) == 0) {
+			if (PIOS_BMP280_Test() != 0) {
+				PIOS_SENSORS_SetMissing(PIOS_SENSOR_BARO);
+			}
+		}
+		else {
+			PIOS_SENSORS_SetMissing(PIOS_SENSOR_BARO);
+		}
+#endif // PIOS_INCLUDE_BMP280
+#endif // PIOS_INCLUDE_I2C
+	}
+
+	if (((hw_i2c == HWBRAINRE1_I2CPORT_EXTBARO) || (hw_i2c == HWBRAINRE1_I2CPORT_EXTBAROANDEXTMAG)) &&
```

Comment:
How feasible is it to factor this stuff into PIOS_HAL similar to the ConfigurePort stuff?  Other targets will want this same functionality.

(Might want it to return what external sensors were attempted to be initialized as flags, so that in the future targets can be factored to not use internal versions of the sensors based on this return value-- or so we can factor some of the on-board hardware support to PIOS_HAL).